move Libdvd to depends

wsnipex opened the pull request:
this moves libdvd to depends, pulling from our own repositories. 

I gave it a quick try on linux with a test dvd iso, but it gives an error.

> 19:17:12 T:140592437192448    INFO:   msg: libdvdnav: Using dvdnav version 5.0.4
> 19:17:12 T:140592437192448 WARNING: dll_ioctl - 21392 request failed with error [25] Inappropriate ioctl for device
> 19:17:12 T:140592437192448    INFO:   msg: libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
> 19:17:12 T:140592437192448    INFO:   msg: libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed

 No idea if the problem is the (possibly missing) patches in our repo, my changes or something else completely. In any case, this still needs work and testing on all platforms.

@Voyager1 @FernetMenta @afedchin 
@fetzerch fyi, this breaks cmake once again. Unless you beat me to it, I'll take care once this works with autotools.

wsnipex committed:
[depends] add libdvdread

wsnipex committed:
[depends] add libdvdnav

wsnipex committed:
[depends] add libdvdcss

wsnipex committed:
[posix] move libdvd to depends

Voyager1 commented:
@wsnipex the file open error is also known on Windows. It works fine with real local FS, anything to be accessed through VFS seems broken. See https://github.com/xbmc/xbmc/pull/9120#issuecomment-183948666

stefansaraev commented on lib/libdvd/Makefile.in:
```diff
@@ -3,80 +3,89 @@ ARCH=@ARCH@
+	[ -d libdvdcss ] || mkdir libdvdcss
+	cd libdvdcss; ar x $(DVDCSS_A)
+	cd libdvdnav; ar x $(DVDNAV_A)
+	cd libdvdread; ar x $(DVDREAD_A)
```
well. after moving to depends, aren't those supposed to be linked like -ldvdcss -ldvdnav -lfoo 
why caling into depends from here and then unpacking the archives ?

notspiff commented:
doing this while relying on old wrapping hacks is .. suboptimal. do it properly if you want to do it.

wsnipex commented:
trust me, I wanted to get rid of this crap and just statically link, libs are small after all. 
@FernetMenta wanted to keep it dyloaded and this is the result. Getting the wrapper working for cross and native compile actually took the better part of the day...

I'm not going to rewrite core code I have no knowledge in for a feature I hardly use myself.
Any fixes would be greatly appreciated.

FernetMenta commented:
@wsnipex thanks! I will have a closer look this evening. Sure, in the long term it makes more sense to get rid of the hacks but for the time being this blocks msys2 upgrade. I think it makes more sense to move forward in iterations, means keeping libdvd as it was (just upgrade). In a later iteration we get drop the wrapper.

ace20022 commented:
@Voyager1 @FernetMenta I want to replace our udf impl with libudfread anyway, so I could look at the callbacks too, but I can't give a date. I just want to avoid that 3 devs work at the same thing.

Voyager1 commented:
@ace20022 - I'd like to get libdvd 5.0.4 working with the old wrapper and msys2 if possible (I think it's compiler/linker settings related). But I'm not going to touch the callbacks thing. So please feel free ...

akva2 commented:
the main problem is that the hook in libdvd is compile time only. it's trivial https://github.com/xbmc/libdvdcss/blob/master/src/device.c#L77 to hook in our io but it cannot be done for precompiled libraries.

wsnipex commented:
jenkins build this please

wsnipex commented:
updated. I noticed that tarballs of git branches were never re-downloaded, so updates to the repos were ignored. Fixed by resolving the git hash and using this in the filename. @afedchin you'll probably need something similar for win32.
jenkins build this please

wsnipex commented:
jenkins build this please

MartijnKaijser commented:
jenkins build this please

afedchin commented:
@wsnipex any news on this?

wsnipex commented:
doesn't build on IOS yet(as usual), cause it lacks ioctls dvdcss needs. That means I have to reintroduce an option to disable it, which I really wanted to avoid.

wsnipex commented:
jenkins build this please

wsnipex commented:
jenkins build this please

wsnipex commented:
sigh, jenkins build this please

wsnipex commented:
jenkins build this please

wsnipex commented:
ok, its finally working. I only tested linux though.

FernetMenta commented:
great! I will test on OSX now

wsnipex commented:
any test results?

FernetMenta commented:
looks good on OSX

wsnipex commented:
I guess we should also test droid and ios, but up to you.
Your button @FernetMenta and @afedchin

MartijnKaijser commented:
can do some tests tonight on droid

MartijnKaijser commented:
fails to build on win32 at least

wsnipex commented:
yes, cause the win32 part is https://github.com/xbmc/xbmc/pull/9120

FernetMenta commented:
@wsnipex thanks very much!
@afedchin I think it is best if you merge this PR with yours

afedchin commented:
@FernetMenta oki, will cherry-pick this a little later.

MartijnKaijser commented:
all fine on android with DVD menus

afedchin commented:
@FernetMenta all done. #9120 was updated. Jenkins is happy.

FernetMenta commented:
inject :)

afedchin commented:
close this in favor of #9120
